### DISCUSSION TITLE
Consider saving entry revisions on the file system, instead of database

### DISCUSSION THREAD

**Discussion by litzinger** [ExpressionEngine/ExpressionEngine#3311] [Feature Requests]
The exp_entry_versioning table, in large sites, can often account for half or more than half of the overall database size. For example we have one site that has 500+ mb database, but 250mb of it is the exp_entry_versioning table alone.

Now that Flysystem is native to EE 7, it should be easier to implement such a feature. Could possibly keep the exp_entry_versioning table as is, but instead of saving the JSON in the version_data table, it could store a reference to a file on S3 (or another file system)

- [x] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by swierczek** [comment]
Another idea was tossed around in Slack (from October 2023, though lost to history at this point) - to directly address the size of this table, it might be possible to use MySQL's COMPRESS/UNCOMPRESS functions to write/read the data. Doing so would remove the ability to easily query the table looking for specific save data. But that's a pretty minimal downside though, because I'd think that's a very rare need.
